qdigidoc: 4.2.11 -> 4.2.12

flokli opened the pull request:
From https://github.com/open-eid/DigiDoc4-Client/releases/tag/v4.2.12:

> Code, Text and translation improvements and updates


cc @klemensn 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

flokli committed:
qdigidoc: 4.2.11 -> 4.2.12

From https://github.com/open-eid/DigiDoc4-Client/releases/tag/v4.2.12:

> Code, Text and translation improvements and updates

klemensn commented on pkgs/tools/security/qdigidoc/default.nix:
```diff
@@ -4,12 +4,12 @@
-      "https://github.com/open-eid/DigiDoc4-Client/releases/download/v${version}/qdigidoc4_${version}.110-1804.tar.xz";
-    sha256 = "sha256-Sg6lFZeIJn3T/suDc5Z/kNqBf/sIV9c6EJJ0Nr0dwTM=";
+      "https://github.com/open-eid/DigiDoc4-Client/releases/download/v${version}/qdigidoc4_${version}.118-1804.tar.xz";
+    hash = "sha256-tdirhSEpBa1hnpau+bWP9Ioy44DLJHulSBx7xWl0hG4=";
```
Why not https://github.com/open-eid/DigiDoc4-Client/releases/download/v4.2.12/qdigidoc4-4.2.12.tar.gz?
What are the differences between these tarballs?

flokli replied:
I don't know - this was the download URL that was used in previous iterations of the derivations.

It seems the other tarball also builds without problems, so I'll use that one.

klemensn commented:
It might be worth cherry-picking some PRs until the next release.

flokli committed:
qdigidoc: nixpkgs-fmt

flokli committed:
qdigidoc: use other source

It seems there's another version of the tarball published, without a
magic number that needs to be manually incremented. Use that.

flokli commented:
> It might be worth cherry-picking some PRs until the next release. At least [open-eid/digidoc-extensions#21](https://github.com/open-eid/digidoc-extensions/pull/21) comes to mind.

No, I prefer not doing upstreams job and maintaining our own distribution of qdigidoc ;-) nixpkgs should only carry nixpkgs-specific patches ideally.
